types/ember-qunit/v2/index.d.ts

```diff
@@ -48,66 +48,13 @@ export class QUnitAdapter extends Ember.Test.Adapter {}
 export { module, test, skip, only, todo } from 'qunit';
 
 declare global {
-    interface QUnit {
-        /**
-         * Add a test to run.
-         *
-         * Add a test to run using `QUnit.test()`.
-         *
-         * The `assert` argument to the callback contains all of QUnit's assertion
-         * methods. Use this argument to call your test assertions.
-         *
-         * `QUnit.test()` can automatically handle the asynchronous resolution of a
-         * Promise on your behalf if you return a thenable Promise as the result of
-         * your callback function.
-         *
-         * @param name Title of unit being tested
-         * @param callback Function to close over assertions
-         */
-        test(name: string, callback: (this: TestContext, assert: Assert) => void): void;
-
-        /**
-         * Adds a test to exclusively run, preventing all other tests from running.
-         *
-         * Use this method to focus your test suite on a specific test. QUnit.only
-         * will cause any other tests in your suite to be ignored.
-         *
-         * Note, that if more than one QUnit.only is present only the first instance
-         * will run.
-         *
-         * This is an alternative to filtering tests to run in the HTML reporter. It
-         * is especially useful when you use a console reporter or in a codebase
-         * with a large set of long running tests.
-         *
-         * @param name Title of unit being tested
-         * @param callback Function to close over assertions
-         */
-        only(name: string, callback: (this: TestContext, assert: Assert) => void): void;
-
-        /**
-         * Use this method to test a unit of code which is still under development (in a “todo” state).
-         * The test will pass as long as one failing assertion is present.
-         *
-         * If all assertions pass, then the test will fail signaling that `QUnit.todo` should
-         * be replaced by `QUnit.test`.
-         *
-         * @param name Title of unit being tested
-         * @param callback Function to close over assertions
-         */
-        todo(name: string, callback: (this: TestContext, assert: Assert) => void): void;
-
-        /**
-         * Adds a test like object to be skipped.
-         *
-         * Use this method to replace QUnit.test() instead of commenting out entire
-         * tests.
-         *
-         * This test's prototype will be listed on the suite as a skipped test,
-         * ignoring the callback argument and the respective global and module's
-         * hooks.
-         *
-         * @param Title of unit being tested
-         */
-        skip: typeof QUnit.skip;
+    interface TestOnly {
```

Comment:
We shouldn't back-port this to the old ember-qunit v2 types, I think. It's long out of support and I'm not actually sure it would even actually work correctly. Did you *need* to do that to get tests passing? 🤔

Reply:
Yeah. I didn't originally as I only tested `qunit` plus `ember-qunit` locally and noticed the failing CI. I was able to reproduce it via `npm test ember-qunit/v4`.

@chriskrycho Could it be due to `import from 'qunit'` resolving to the latest? Hence the same limitation/conflict is being found here as well.

Reply:
Probably. I'll follow up with the DT maintainers early next week and see if there's a way to make this work without bumping those but it's not the end of the world if not.

Reply:
I’m not quite following, what’s the issue/question?

Reply:
Ah, apologies, I forgot to come back to this. The problem is that we have old (now very much unsupported!) versions of `ember-qunit` which depend on the types for `qunit`, and while they should *probably* work with these updated types for `qunit`, I have no idea how that would actually play out in practice with the runtime version of the libraries. Net, I would *prefer* to leave the types for `ember-qunit` `v2` and `v4` resolving to the *previous* version of the types for `qunit`, and only bump the types relationship for `ember-qunit` v5/v6 (i.e. the current latest). I don't know if there's a way to do that, though!

(I am, separately, chipping away at getting all of Ember's types published directly which would solve this in its own right, and is undoubtedly the *best* solution here.)

Reply:
You can add `@types/qunit` to [here](https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/definitions-parser/allowedPackageJsonDependencies.txt) and then pin the version of `@types/qunit` in `types/ember-qunit/v2/package.json`.

Reply:
Ah, thank you! @krinkle, if you wouldn't mind doing that, it would be grand. (I'm also going to keep pushing on ember-unit publishing its own types, though probably it won't be till late this week or early next at earliest.)